web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -0,0 +1,341 @@
+   */
+  private initialChart(): void {
+    const element = this.barChartContainer.nativeElement;
+    const elementWidth = element.clientWidth;
+    const elementHeight = element.clientHeight;
+    const marginFocus = { top: 60, right: 90, bottom: 150, left: 70 };
+    const marginBrush = { top: 370, right: 60, bottom: 20, left: 60};
+
+    this.heightFocus = elementHeight;
+    this.heightBrush = 30;
+    this.width = elementWidth;
+
+    this.svg = (d3
+      .select(element)
+      .append('svg') as d3SVG)
+      .attr('width', elementWidth)
+      .attr('height', elementHeight);
+
+    this.focus = this.svg
+      .append('g')
+      .attr('transform','translate(0, 0)');
+    this.brush = this.svg
+      .append('g')
+      .attr('transform', `translate(0, ${marginBrush.top})`);
+
+    this.xScaleFocus = d3
+      .scaleBand()
+      .rangeRound([marginFocus.left, elementWidth - marginFocus.right])
+      .padding(0.1);
+    this.yScaleFocus = d3
+      .scaleLinear().range([elementHeight - marginFocus.bottom, marginFocus.top]);
+    this.xAxisFocus = this.focus
+      .append('g')
+      .attr('class', 'axis axis--x')
+      .attr('transform', `translate(0, ${elementHeight - marginFocus.bottom})`);
+    this.yAxis = this.focus
+      .append('g')
+      .attr('transform', `translate(${marginFocus.left}, 0)`)
+      .style('color', BarChartComponent.COLOR_LIGHT_GRAY);
+
+    this.focus.append("text")
+      .attr("text-anchor", "middle") 
+      .attr("transform", 
+        "translate("+(marginFocus.left/2)+","+
+        ((elementHeight-marginFocus.bottom+marginFocus.top)/2)+")rotate(-90)")
+      .attr("fill", BarChartComponent.COLOR_LIGHT_GRAY)
+      .text("Duration of the push (hours)")
+      .style("font", "12px sans-serif");
+
+    this.xScaleBrush = d3
+      .scaleBand()
+      .rangeRound([marginBrush.left, elementWidth - marginBrush.right])
+      .padding(0.1)
+    this.yScaleBrush = d3.scaleLinear().range([this.heightBrush, 0]);
+    this.xAxisBrush = this.brush.append('g')
+      .attr('transform', `translate(0, ${this.heightBrush})`);
+  }
+
+  /**
+   * This function updates the focus barchart based on the data of the 
+   * dropdown selection.
+   * @param dataSelection: Value selected for the bar chart
+   */
+  public updateChart(dataSelection: number): void {
+    let dataSelected = []
+    if (dataSelection === 0) {
+      dataSelected = this.dataAll;
+    } else {
+      dataSelected = this.dataComplete;
+    }
+    if (!dataSelected) { return; }
+
+    const maxDuration = d3.max(dataSelected, (d: BarChartData) => d.duration);
+    if (!maxDuration) { return; }
+
+    // Upate the xScaleFocus, yScaleFocus, xAxisFocus and yAxisFocus
+    // based on the selected data for the focus chart.
+    this.xScaleFocus.domain(dataSelected.map((d: BarChartData) => d.startTime));
+    this.yScaleFocus.domain([0, maxDuration]);
+    this.yAxis.transition().duration(0)
+      .call(d3.axisLeft(this.yScaleFocus));
```

Comment:
I feel like this can go on the previous line and it'll look fine, especially since you already chain `transition` and `duration` together on the same line